src/components/TodoFilter/TodoFilter.tsx

```diff
@@ -1,34 +1,71 @@
-export const TodoFilter = () => (
-  <form className="field has-addons">
-    <p className="control">
-      <span className="select">
-        <select data-cy="statusSelect">
-          <option value="all">All</option>
-          <option value="active">Active</option>
-          <option value="completed">Completed</option>
-        </select>
-      </span>
-    </p>
-
-    <p className="control is-expanded has-icons-left has-icons-right">
-      <input
-        data-cy="searchInput"
-        type="text"
-        className="input"
-        placeholder="Search..."
-      />
-      <span className="icon is-left">
-        <i className="fas fa-magnifying-glass" />
-      </span>
-
-      <span className="icon is-right" style={{ pointerEvents: 'all' }}>
-        {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-        <button
-          data-cy="clearSearchButton"
-          type="button"
-          className="delete"
+import { FC, useState } from 'react';
+
+import { Todo } from '../../types/Todo';
+
+type Props = {
+  search: (value: string, select: string) => Todo[] | null;
+  setVisibleTodos: (val: Todo[] | null) => void;
+};
+
+export const TodoFilter:FC<Props> = ({ search, setVisibleTodos }) => {
+  const [text, setText] = useState('');
+  const [selected, setSelected] = useState('all');
+
+  return (
+    <form className="field has-addons">
+      <p className="control">
+        <span className="select">
+          <select
+            data-cy="statusSelect"
+            value={selected}
+            onChange={(event) => {
+              setSelected(event.target.value);
+              const todos = search(text, event.target.value);
+
+              setVisibleTodos(todos);
```

Comment:
do u need this state, as for me it only creates more rerenders😅

Reply:
Sorry, I didn`t understand your tip, what exactly state is unneeded?

Reply:
You can use useMemo instead of using setState to remove a couple of redundant renderers, but it's optional